backend/src/main/java/com/empire/Nation.java

```diff
@@ -63,6 +63,21 @@ public boolean hasTag(NationData.Tag tag) {
 			return trait1 == tag || trait2 == tag;
 		}
 
+		public boolean hasScoreProfile(NationData.ScoreProfile profile) {
```

Comment:
This string representation and hardcoded score profile class members seems kind of clunky to me.  Is there a reason not to use the same set-based pattern you've been using for all the various Tags in other classes?  If not, I this PR would be the appropriate time to make this change.

Reply:
I hear you, not happy about this either. The reason is that this is parsed from form-to-json directly, and the form to json logic provides a completely flat object. So the options are to sort of handle it like orders (big map of strings -> strings) or like it is here (explicit members), or to build some adapter from one or the other to more structured data. setup.html is simple enough that we could manually build some structured json from the form elements instead of using the library, but that is a larger change.